ArduPlane/quadplane.cpp

```diff
@@ -1360,6 +1360,12 @@ void QuadPlane::set_armed(bool armed)
     if (plane.control_mode == &plane.mode_guided) {
         guided_wait_takeoff = armed;
     }
+
+    // re-init throttle wait on arm and disarm, to prevent rudder
+    // arming on 2nd flight causing yaw
+    if (!air_mode_active()) {
```

Comment:
I don't think this works in the arm and disarm with airmode case, there we arm and then set airmode.

Reply:
i need to add the air mode check to pass the air mode autotest. The autotest shows that motors start with no throttle up, at least in QACRO and QSTABILIZE. That does make sense with arming on a switch I think, which is what is generally used in with airmode

Reply:
maybe you missed the change that resets throttle_wait after airmode is set?